django/db/models/fields/related_objects.py

```diff
@@ -0,0 +1,315 @@
+"""
+"Rel objects" for related fields.
+
+"Rel objects" (for lack of a better name) carry information about the relation
+modeled by a related field and provide some utility functions. They're stored
+in the ``rel`` attribute of the field.
```

Comment:
We will most likely need ReverseManyToManyField, at least in the short term (for example, one gets migrations, the other doesnt').

I'm fine with both one file or split to two files. That is, implementor's decision is fine for me.